javaslang/src/main/java/javaslang/collection/Seq.java

```diff
@@ -1153,4 +1149,38 @@ default boolean startsWith(Iterable<? extends T> that, int offset) {
     default Function1<Integer, T> withDefault(Function<? super Integer, ? extends T> defaultFunction) {
         return i -> (i >= 0 && i < length()) ? apply(i) : defaultFunction.apply(i);
     }
+
+    default java.util.List<T> asUnmodifiableList() { return new SeqList<>(Seq.this); }
+
+    @SuppressWarnings("unchecked")
+    class SeqList<T> implements java.util.List<T> {
+        private final Seq<T> delegate;
+        SeqList(Seq<T> delegate) { this.delegate = delegate;}
+
```

Comment:
this formatting actually helps in the `IDE`: nothing to see here, all method are simply delegated!
https://github.com/paplorinc/javaslang/blob/62915babab93ee2d7d0d0e0373bf3f435fb6d4ac/javaslang/src/main/java/javaslang/collection/Seq.java#L1160